packages/wrangler/scripts/deps.ts

```diff
@@ -22,6 +22,7 @@ export const EXTERNAL_DEPENDENCIES = [
 	// and read when we are bundling the worker application
 	"unenv",
 	"workerd/worker.mjs",
+	"@cloudflare/workers-shared/dist",
```

Comment:
marking `@cloudflare/workers-shared/dist` in esbuild as [external](https://esbuild.github.io/api/#external) means that "instead of being bundled, the import will be preserved and will be evaluated at run time instead.". This unfortunately comes with the side effect that `@cloudflare/workers-shared/dist` won't be considered a dependency, and will therefore be ignored in watch mode. This means that when in `dev` mode, any changes to the `asset-server-worker` code (or any code inside `@cloudflare/workers-shared` for that matter) won't trigger a wrangler re-bundle, which we'd ideally want, since `miniflare` references the bundled ASW [here](https://github.com/cloudflare/workers-sdk/pull/6370/files#diff-ca7f0876a94f845d3a9323b32cba14fbfb3add60d26b7a0d86a5fc68beb3ca3aR846).

While this will most likely not impact end-users, since they shouldn't be fiddling with `workers-shared/ASW` anyway, it does create some friction for us as the wrangler team, and our dev experience.

One possible solution I played with, that seemed to do the trick, was to factor the rebuilding of ASW in the wrangler dev process, meaning 👇 (with `pnpm -C ../workers-shared run bundle:asset-server --watch` being the bit of interest here)
```
# packages/wrangler/package.json

"dev": "pnpm run clean && concurrently -c black,blue --kill-others-on-fail false \"pnpm -C ../workers-shared run bundle:asset-server --watch\" \"pnpm run bundle --watch\" \"pnpm run check:type --watch --preserveWatchOutput\""
```

But tbh, I'm not sure how I feel abt that, and since I feel like I'd like to have validation of this work **before** I spend more time on this, I left it out of my PR for the time being